index.js

```diff
@@ -82,37 +83,25 @@ function HttpServer({ listeningPort, rootFolder, sslConfig }, callback) {
 
 		console.log(`Checking if port ${port} is available. Please wait...`);
 
-		require(commType).request({ port }, (res) => {
-			callback(undefined, true);
-		}).on('error', (err) => {
+		const req = require(commType).request({ port }, (res) => {
+			res.on('data', (_) => {});
+			res.on('end', () => {
+				callback(undefined, true);
+			})
+		});
+		req.on('error', (err) => {
 			callback(undefined, false);
 		});
-	}
 
-	function bindErrorHandler(error) {
-		if (error.code === 'EADDRINUSE') {
-			server.close();
-			if (callback) {
-				return callback(error);
-			}
-			throw error;
-		}
+		req.end();
 	}
 
-	function bindFinished(err) {
-		if (err) {
-			console.log(err);
-			if (callback) {
```

Comment:
The `callback` parameter cannot be passed so there's no reason keep it

Reply:
daca imi aduc bine aminte, se dadea callback cand se incearca ridicarea unui APIHUB din TIR (TestIntegrationTesting) si exista un mechanism care tot incerca sa ii dea porturi random lui apihub pe care sa porneasca.